src/common/commonutils/UrlUtils.c

```diff
@@ -32,7 +34,7 @@ char* UrlEncode(char* target)
             }
             else
             {
-                sprintf(&encodedTarget[j], "%%%02X", target[i]);
+                snprintf(&encodedTarget[j], encodedLength - j, "%%%02X", target[i]);

```

Comment:
If we need the length parameter here, it means we've allocated a wrong buffer size. This in fact may be worse than crashing, because we'll silently return an ill-formed url.

Reply:
Agree. Back to sprintf we go.